### DISCUSSION TITLE
Welcome to Telego Discussions!

### DISCUSSION THREAD

**Discussion by mymmrac** [mymmrac/telego#12] [Announcements]
# 👋 Welcome everyone!

Discussions used as a place to connect with members of community. Hope that you:

- Ask questions you’re wondering about
- Share ideas
- Engage with other community members
- Welcome others and are open-minded

🙂 Have fun using Telego and share your thoughts with everyone.

**Comment by sarff** [comment]
Hi. Could you help with ForceReply? if i have like this, how i can send message ("May I know your age, please?") with Force reply?:
```
tlg.Bh.HandleCallbackQuery(func(bot *telego.Bot, query telego.CallbackQuery) {
		// Send message
		_, _ = bot.SendMessage(tu.Message(tu.ID(query.Message.Chat.ID), "May I know your age, please?"))

		// Answer callback query
		_ = bot.AnswerCallbackQuery(tu.CallbackQuery(query.ID).WithText("Done"))

	}, th.AnyCallbackQueryWithMessage(), th.CallbackDataEqual("letsgo_questionnaire"))
```
thx a lot

**Reply by sarff** [reply]
If I understand correctly I have one way:

```
tlg.Bh.HandleCallbackQuery(func(bot *telego.Bot, query telego.CallbackQuery) {
		// Send message
		//_, _ = bot.SendMessage(tu.Message(tu.ID(query.Message.Chat.ID), "May I know your age, please?"))

		r := &telego.SendMessageParams{
			ChatID:      tu.ID(query.Message.Chat.ID),
			ReplyMarkup: &telego.ForceReply{ForceReply: true},
			Text:        "May I know your age, please?",
		}
		_, _ = bot.SendMessage(r)

		// Answer callback query
		_ = bot.AnswerCallbackQuery(tu.CallbackQuery(query.ID).WithText("Done"))

	}, th.AnyCallbackQueryWithMessage(), th.CallbackDataEqual("letsgo_questionnaire"))
```

**Reply by mymmrac** [reply]
You can do that with reply markup option, like this:
```go
tu.Message(tu.ID(query.Message.Chat.ID), "May I know your age, please?").
	WithReplyMarkup(&telego.ForceReply{
		ForceReply:            true,
		InputFieldPlaceholder: "Age",
	}),
```

**Reply by mymmrac** [reply]
Telego's `ReplyMarkup` is an interface that is implemented by [InlineKeyboardMarkup](https://core.telegram.org/bots/api#inlinekeyboardmarkup), [ReplyKeyboardMarkup](https://core.telegram.org/bots/api#replykeyboardmarkup), [ReplyKeyboardRemove](https://core.telegram.org/bots/api#replykeyboardremove), [ForceReply](https://core.telegram.org/bots/api#forcereply) types

**Comment by sarff** [comment]
Hi. If I am using a multi bot webhook, how can I tell which bot the update came from (for example if it is a contact transfer)? Thanks.

**Reply by mymmrac** [reply]
Hi, generally you would want to have webhook configured for different paths, so that updates from different bot's are separated

From [multi bot example](https://github.com/mymmrac/telego/blob/main/examples/multi_bot_webhook/main.go):
```go
updates1, _ := bot1.UpdatesViaWebhook(
  "/bot"+bot1.Token(),
  telego.WithWebhookServer(srv),
)
updates2, _ := bot2.UpdatesViaWebhook(
  "/bot"+bot2.Token(),
  telego.WithWebhookServer(srv),
)
```

Updates will come into two different URLs (`/bot_TOKEN_1` and `/bot_TOKEN_2`)

If you want to have your updates on a single URL, I guess it will be not so easy to do with Telego right now, and actually I suppose with any other library since there is not much info in Telegram updates to differentiate them

**Reply by sarff** [reply]
So I can determine where the message came from by the entry point https://example.com/tlg1 or tlg2.  
But to determine message.frombot_id - there is nothing like that? right?

**Reply by mymmrac** [reply]
Yep, this is the limitation of Telegram's updates, they generally not include any info about bot
